crates/uv-distribution-types/src/dist_error.rs

```diff
@@ -1,8 +1,69 @@
-use std::fmt::{Display, Formatter};
+use std::fmt::{Debug, Display, Formatter};
+use std::ops::Deref;
 use uv_normalize::{ExtraName, GroupName, PackageName};
 use uv_pep440::Version;
 use version_ranges::Ranges;
 
+/// Orphan-rule workaround for inspecting build frontend errors.
+///
+/// Normally we would match on the variants, but we have to pass this error through an opaque type
+/// to avoid cyclical crate dependencies between the build frontend and resolver/installer.
+pub trait IsBuildBackendError: std::error::Error + Send + Sync + 'static {
+    /// Returns whether the build backend failed to build the package, so it's not a uv error.
+    fn is_build_backend_error(&self) -> bool;
+}
+
+/// Wrapper type to make `thiserror`'s `AsDynError` work with `IsBuildFrontendError`.
```

Comment:
Do you mean `IsBuildBackendError`?

Reply:
Also, I would mention that `AsDynError` is an internal thing in `thiserror`. It took me a bit to figure that out as I was confused about it not being in `thiserror`'s public API docs.